contracts/external/cover/CoverBroker.sol

```diff
@@ -0,0 +1,194 @@
+
+  // Immutables
+  ICover public immutable cover;
+  IMemberRoles public immutable memberRoles;
+  IPool public immutable pool;
+  INXMToken public immutable nxmToken;
+  address public immutable tokenController;
+
+  // Constants
+  address public constant ETH = 0xEeeeeEeeeEeEeeEeEeEeeEEEeeeeEeeeeeeeEEeE;
+  uint private constant ETH_ASSET_ID = 0;
+  uint private constant NXM_ASSET_ID = type(uint8).max;
+
+  constructor(address _cover, address _memberRoles, address _pool, address _nxmToken, address _tokenController) {
+    cover = ICover(_cover);
+    memberRoles = IMemberRoles(_memberRoles);
+    pool = IPool(_pool);
+    nxmToken = INXMToken(_nxmToken);
+
+    tokenController = _tokenController;
+    nxmToken.approve(tokenController, type(uint256).max);
+  }
+
+  /// @notice Buys cover on behalf of the caller. Supports ETH, NXM and ERC20 asset payments which are supported by the pool.
+  /// @dev For NXM and ERC20 payments, ensure the Cover contract is approved to spend the tokens first (maxApproveCoverContract).
+  /// See supported ERC20 payment methods via pool.getAssets.
+  /// @param params The parameters required to buy cover.
+  /// @param poolAllocationRequests The allocation requests for the pool's liquidity.
+  /// @return coverId The ID of the purchased cover.
+  function buyCover(
+    BuyCoverParams calldata params,
+    PoolAllocationRequest[] calldata poolAllocationRequests
+  ) external payable returns (uint coverId) {
+
+    // ETH payment
+    if (params.paymentAsset == ETH_ASSET_ID) {
+      return _handleEthPayment(params, poolAllocationRequests);
+    }
+
+    // msg.value must be 0 if not an ETH payment
+    if (msg.value > 0) {
+      revert InvalidPayment();
+    }
+
+    // NXM payment
+    if (params.paymentAsset == NXM_ASSET_ID) {
+      return _handleNxmPayment(params, poolAllocationRequests);
+    }
+
+    // ERC20 payment
+    return _handleErc20Payment(params, poolAllocationRequests);
+  }
+
+  /// @notice Allows the Cover contract to spend the maximum possible amount of a specified ERC20 token on behalf of the CoverBroker.
+  /// @param erc20 The ERC20 token for which to approve spending.
+  function maxApproveCoverContract(IERC20 erc20) external onlyOwner {
+    erc20.safeApprove(address(cover), type(uint256).max);
+  }
+
+  /// @notice Switches CoverBroker's membership to a new address.
+  /// @dev MemberRoles contract needs to be approved to transfer NXM tokens to new membership address.
+  /// @param newAddress The address to which the membership will be switched.
+  function switchMembership(address newAddress) external onlyOwner {
+    nxmToken.approve(address(memberRoles), type(uint256).max);
+    memberRoles.switchMembership(newAddress);
+  }
+
+  /// @notice Transfers all available funds of a specified asset (ETH or ERC20) to the contract owner.
+  /// @param assetAddress The address of the asset to be transferred.
+  function transferFunds(address assetAddress) external onlyOwner {
```

Comment:
I think we could rename to maybe `rescueAsset` to make the purpose of the function more clear. Up to you to decide if it's better or not.

Reply:
i like `rescueAsset`